platform-sdk/swirlds-base/src/main/java/com/swirlds/base/utility/ToStringBuilder.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.base.utility;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.Objects;
+
+/**
+ * A builder for toString methods.
+ */
+public class ToStringBuilder {
+    private final StringBuilder builder;
+    private final Class<?> thisClass;
+    private final Class<?> superClass;
+    private final int initLength;
+
+    public ToStringBuilder(@NonNull final Object object) {
+        builder = new StringBuilder();
+        thisClass = object.getClass();
+        superClass = thisClass.getSuperclass();
+
+        builder.append(thisClass.getSimpleName()).append("[");
+        initLength = builder.length();
+    }
+
+    private ToStringBuilder(@NonNull final String superString) {
```

Comment:
I'm not understanding the necessity of this alternate constructor. As far as I can tell, the way it's being used in line 53 is just returning `superString` exactly

Reply:
Need the `toString` method of the ne `ToStringBuilder`

Reply:
I'm not sure what you mean - if I delete this constructor and modify line 53 to be

```
builder.append(replacedString).append(",");
```

then all the tests still pass. `toString` isn't doing anything with the input string

Reply:
some rewrite happed :)